libresin/libresin/core/id_registry.hpp

```diff
@@ -0,0 +1,141 @@
+#ifndef RESIN_COMPONENTS_HPP
+#define RESIN_COMPONENTS_HPP
+#include <functional>
+#include <glm/fwd.hpp>
+#include <glm/glm.hpp>
+#include <libresin/utils/exceptions.hpp>
+#include <libresin/utils/logger.hpp>
+#include <mutex>
+#include <stack>
+
+namespace resin {
+
+// Thread safe id manager.
```

Comment:
Do we plan to employ multi-threading or did you just do it for fun? 😺

Reply:
I did for fun xd, mostly because the id_registry was a singleton at the beginning. I've decided to change it to a regular class, as it's more modular and it allows for creation of many trees at the same time. 

It would be good to remove the mutex then as we are paying a redundant cost